### PR TITLE
Modify GET request to receive user's boards

### PR DESCRIPTION
**Pull request by Szachista**

### CODE REVIEW

File: src/test/java/com/intive/patronage22/szczecin/retroboard/controller/BoardControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        user1.setUid("def");
+        user1.setName("Test user 1");
+        userRepository.save(user1);
+
+        Board b = new Board();
+        b.setName("Test board 1");
+        b.setCreator(user1);
+        boardRepository.save(b);
+
+        Board b2 = new Board();
```
Comment (Radenzvous) [inline src/test/java/com/intive/patronage22/szczecin/retroboard/controller/BoardControllerTest.java:60]: Can we have the same naming convention as for users?  board1, board2

File: src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java [hunk 2]
```diff
@@ -1,26 +1,47 @@
...
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
 import java.util.ArrayList;
 import java.util.List;
 
 
+@Transactional
```
Comment (MaciejJ0) [inline src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java:16]: Better to use @Transactional for methods

File: src/main/java/com/intive/patronage22/szczecin/retroboard/controller/BoardController.java [hunk 3]
```diff
@@ -4,26 +4,32 @@
...
-
+    public ResponseEntity<List<BoardDto>> getUserBoards(@RequestParam(name = "userId", required = false) String uid) {
+        if (uid == null || uid.isBlank()) {
+            final List<BoardDto> boards = boardService.mockBoardData();
+            return ResponseEntity.status(OK).body(boards);
+        }
+        
+        List<BoardDto> boards = boardService.getUserBoards(uid);
+        return ResponseEntity.status(boards == null ? NOT_FOUND : OK).body(boards);
     }
```
Comment (MaciejJ0) [inline src/main/java/com/intive/patronage22/szczecin/retroboard/controller/BoardController.java:34]: It is better to move the logic to the server and return "OK". 
Other answers will be in ControllerAdvice.

File: src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java (hunk 2 above)
Comment (Szachista) [reply src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java]: Done.

File: src/test/java/com/intive/patronage22/szczecin/retroboard/controller/BoardControllerTest.java (hunk 1 above)
Comment (Szachista) [reply src/test/java/com/intive/patronage22/szczecin/retroboard/controller/BoardControllerTest.java]: Done.

File: src/main/java/com/intive/patronage22/szczecin/retroboard/controller/BoardController.java (hunk 3 above)
Comment (Szachista) [reply src/main/java/com/intive/patronage22/szczecin/retroboard/controller/BoardController.java]: Probably done.

File: src/main/java/com/intive/patronage22/szczecin/retroboard/dto/BoardDto.java [hunk 4]
```diff
@@ -1,15 +1,14 @@
...
-import lombok.RequiredArgsConstructor;
-import lombok.Value;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
 
-@Value
-@Getter
-@RequiredArgsConstructor
+@Data
```
[the remark this answers is not in the document]
Comment (Szachista) [reply src/main/java/com/intive/patronage22/szczecin/retroboard/dto/BoardDto.java:7]: I wanted to use ModelMapper - it requires setters. What is a better way to map model to dto?

File: src/test/java/com/intive/patronage22/szczecin/retroboard/controller/ControllerAndServiceIntegrationTest.java [hunk 5]
```diff
@@ -23,13 +26,12 @@
     @SpyBean
     private BoardService service;
 
-    @Test
-    public void statusIsOk_TwoObjectsReturned() throws Exception {
+    @MockBean private UserRepository userRepository;
```
[the remark this answers is not in the document]
Comment (Szachista) [reply src/test/java/com/intive/patronage22/szczecin/retroboard/controller/ControllerAndServiceIntegrationTest.java:29]: In the WebMvcTest I cannot use userRepository, so I made it in the BoardControllerTest (annotated with SpringBootTest).

File: src/main/java/com/intive/patronage22/szczecin/retroboard/controller/BoardController.java [hunk 6]
```diff
@@ -2,26 +2,27 @@
...
 import java.util.List;
 
 import static org.springframework.http.HttpStatus.OK;
 
-@RequiredArgsConstructor
 @RestController
 @RequestMapping("/boards")
 public class BoardController {
-    private final BoardService boardService;
+    @Autowired
```
[the remark this answers is not in the document]
Comment (Szachista) [reply src/main/java/com/intive/patronage22/szczecin/retroboard/controller/BoardController.java]: Restored with @RequiredArgsConstructor

File: src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java [hunk 7]
```diff
@@ -1,20 +1,38 @@
...
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.server.ResponseStatusException;
 
 import java.util.ArrayList;
 import java.util.List;
 
 @Service
 public class BoardService {
+    @Autowired
```
[the remark this answers is not in the document]
Comment (Szachista) [reply src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java]: Done.

File: src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java [hunk 8]
```diff
@@ -1,20 +1,38 @@
...
-    public List<BoardDto> mockBoardData() {
-        final List<BoardDto> boardDTOS = new ArrayList<>();
-        boardDTOS.add(new BoardDto("1a", EnumStateDto.CREATED, "RETRO 1"));
-        boardDTOS.add(new BoardDto("2a", EnumStateDto.VOTING, "RETRO 2"));
+    private final ModelMapper modelMapper = new ModelMapper();
 
-        return boardDTOS;
+    @Transactional
+    public List<BoardDto> getUserBoards(String uid) {
+        User u = userRepository.findById(uid).orElse(null);
```
[the remark this answers is not in the document]
Comment (Szachista) [reply src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java]: Done.

File: src/test/java/com/intive/patronage22/szczecin/retroboard/controller/BoardControllerTest.java [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+        assertTrue(userRepository.count() == userCount + 1);
+        ResponseEntity<List<BoardDto>> res = boardController.getUserBoards("abc");
+        userRepository.delete(user);
+        assertTrue(res.getStatusCode() == HttpStatus.OK);
+        assertTrue(res.hasBody());
+    }
+
+    @Test
+    public void whenUserHasTwoBoards_thenBodyShouldContainTwoBoards() {
+        User user1 = new User();
```
[the remark this answers is not in the document]
Comment (Szachista) [reply src/test/java/com/intive/patronage22/szczecin/retroboard/controller/BoardControllerTest.java]: Done.

**Review by Radenzvous — APPROVED** [review]

File: src/main/java/com/intive/patronage22/szczecin/retroboard/controller/BoardController.java [hunk 10]
```diff
@@ -19,9 +20,9 @@
     private final BoardService boardService;
 
     @GetMapping
```
Comment (patrykzawadzkiintive) [inline src/main/java/com/intive/patronage22/szczecin/retroboard/controller/BoardController.java:22]: Please add something like that:
@ApiResponses(value = {
            @ApiResponse(responseCode = "200", description = "Ok"),
            @ApiResponse(responseCode = "404", description = "User boards not found")
    })

File: src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java [hunk 11]
```diff
@@ -2,18 +2,43 @@
...
 import java.util.List;
+import java.util.stream.Collectors;
 
 @Service
+@RequiredArgsConstructor
 public class BoardService {
+    private final UserRepository userRepository;
+
+    @Transactional
+    public List<BoardDto> getUserBoards(String uid) {
```
Comment (patrykzawadzkiintive) [inline src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java:24]: I think this method should just return user boards, i mean Optional<List<BoardDto>>. If optional will be empty then it means user not exist. 

Throwing exception here is kind of mixing responsibilities of controller & service. Service has it owns logic which could be then reused by other services. Move that exception part to controller class.

**Review by patrykzawadzkiintive — CHANGES_REQUESTED** [review]

File: src/main/java/com/intive/patronage22/szczecin/retroboard/controller/BoardController.java (hunk 10 above)
Comment (Szachista) [reply src/main/java/com/intive/patronage22/szczecin/retroboard/controller/BoardController.java:24]: Done.

**Review by patrykzawadzkiintive — APPROVED** [review]

**Review by golldfish — APPROVED** [review]

**Review by MaciejJ0 — APPROVED** [review]

File: src/main/java/com/intive/patronage22/szczecin/retroboard/model/Board.java [hunk 12]
```diff
@@ -35,4 +37,11 @@
             inverseJoinColumns = {@JoinColumn(name = "uid")}
     )
     private Set<User> users = new HashSet<>();
+
+    public static BoardDto toDto(Board board) {
```
Comment (daniel-kmiecik) [inline src/main/java/com/intive/patronage22/szczecin/retroboard/model/Board.java:41]: This method should be moved to the BoardDto class.

File: src/test/java/com/intive/patronage22/szczecin/retroboard/controller/BoardControllerTest.java [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+        assertTrue(userRepository.count() == userCount + 1);
+        ResponseEntity<List<BoardDto>> res = boardController.getUserBoards("abc");
+        userRepository.delete(user);
+        assertTrue(res.getStatusCode() == HttpStatus.OK);
+        assertTrue(res.hasBody());
+    }
+
+    @Test
+    public void whenUserHasTwoBoards_thenBodyShouldContainTwoBoards() {
+        User user1 = makeUser("def","Test user 1");
```
Comment (daniel-kmiecik) [inline src/test/java/com/intive/patronage22/szczecin/retroboard/controller/BoardControllerTest.java:59]: There's a bit of a mess here. First you create one user, then boards and then users again. You should create all users first, and then save them all to the repository with the "saveAll" method. Finally, you can add users to the boards.

File: src/test/java/com/intive/patronage22/szczecin/retroboard/controller/BoardControllerTest.java [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+
+        board2.getUsers().add(user2);
+        boardRepository.save(board2);
+        
+        ResponseEntity<List<BoardDto>> res = boardController.getUserBoards(user2.getUid());
+        assertTrue(res.getStatusCode() == HttpStatus.OK);
+        assertTrue(res.getBody().size() == 2);
+    }
+
+    private User makeUser(String uid, String name) {
```
[the remark this answers is not in the document]
Comment (Szachista) [reply src/test/java/com/intive/patronage22/szczecin/retroboard/controller/BoardControllerTest.java:86]: Done.

File: src/main/java/com/intive/patronage22/szczecin/retroboard/model/Board.java (hunk 12 above)
Comment (Szachista) [reply src/main/java/com/intive/patronage22/szczecin/retroboard/model/Board.java:41]: Done.

File: src/main/java/com/intive/patronage22/szczecin/retroboard/controller/BoardController.java [hunk 15]
```diff
@@ -19,9 +22,14 @@
...
-    public ResponseEntity<List<BoardDto>> getBoards() {
-        final List<BoardDto> boards = boardService.mockBoardData();
-
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "OK"),
+            @ApiResponse(responseCode = "400", description = "Bad request data"),
+            @ApiResponse(responseCode = "404", description = "User not found")
+    })
+    public ResponseEntity<List<BoardDto>> getUserBoards(
+            @RequestParam(name = "userId", required = false) String uid) {
```
[the remark this answers is not in the document]
Comment (Szachista) [reply src/main/java/com/intive/patronage22/szczecin/retroboard/controller/BoardController.java:31]: Done.

**Pull request merged**